src/Microsoft.AspNet.Authentication.Facebook/FacebookAuthenticationMiddleware.cs

```diff
@@ -37,11 +37,11 @@ public class FacebookAuthenticationMiddleware : OAuthAuthenticationMiddleware<Fa
         {
             if (string.IsNullOrWhiteSpace(Options.AppId))
             {
-                throw new ArgumentException(string.Format(CultureInfo.CurrentCulture, Resources.Exception_OptionMustBeProvided, nameof(Options.AppId)));
+                throw new ArgumentException(string.Format(CultureInfo.CurrentCulture, Resources.Exception_OptionMustBeProvided, nameof(Options)));
```

Comment:
This doesn't look right to me. There's no parameter named `Options` either. Plus, supposing there was, the exception message won't make much sense to the user: what they really need to know is that the `AppId` property wasn't set on the options.